pkg/rke2/serviceaccount.go

```diff
@@ -0,0 +1,64 @@
+package rke2
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func restrictServiceAccount(ctx context.Context, namespace string, cs *kubernetes.Clientset) error {
+	var sa *v1.ServiceAccount
```

Comment:
Generally we want to define this as a normal typed value and then take its address. You can remove the nil check on 29, simplifies line 32

Reply:
Made a regular bool and assigned its pointer to the serviceaccount